src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -42,11 +43,28 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             {
                 throw new ArgumentNullException("input");
             }
+
             error = null;
 #if CORECLR
             object obj = null;
             try
             {
+                // JsonConvert.DeserializeObject does not throw an exception when an invalid Json array is passed.
+                // This issue is being tracked by https://github.com/JamesNK/Newtonsoft.Json/issues/1321.
+                // To work around this, we need to identify when input is a Json array, and then try to parse it via JArray.Parse().
+
+                // If input starts with '[' or ends with ']' (ignoring white spaces).
+                if ((Regex.Match(input, @"(^\s*\[)|(\s*\]$)")).Success)
```

Comment:
The reg expression used here seems not right ... @Francisco-Gamino let's discuss it tomorrow.